version: '3.9'

services:
  server:
    container_name: server
    build:
      context: .
    working_dir: /app
    command: uvicorn application:app --host 0.0.0.0 --reload
    ports:
      - "8000:8000"
    env_file:
      - .env
    volumes:
      - .:/app
    depends_on:
      - qdrant
    restart: on-failure
    networks:
      - app_network

  db:
    container_name: postgresql_db
    image: postgres:16-alpine
    restart: always
    user: postgres
    env_file:
      - .env
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=password
      - POSTGRES_DB=docker_embedding_db
      - POSTGRES_LOGGING_COLLECTOR=on
      - POSTGRES_LOG_CONNECTIONS=on
    ports:
      - "5433:5432"
    expose:
      - "5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready" ]
      interval: 1s
      timeout: 5s
      retries: 10
    networks:
      - app_network

  qdrant:
    container_name: qdrant_db
    image: qdrant/qdrant
    ports:
      - "6333:6333"
      - "6334:6334"
    volumes:
      - qdrant_storage:/qdrant/storage
    networks:
      - app_network

volumes:
  qdrant_storage:
  postgres_data:

networks:
  app_network:
    driver: bridge
